lib/webauthn/fake_authenticator.rb

```diff
@@ -50,12 +50,19 @@ def get_assertion(
       user_verified: false,
       aaguid: AuthenticatorData::AAGUID,
       sign_count: nil,
-      extensions: nil
+      extensions: nil,
+      allow_credentials: nil
     )
       credential_options = credentials[rp_id]
 
       if credential_options
-        credential_id, credential = credential_options.first
+        allow_credentials ||= credential_options.keys
+        credential_id = (credential_options.keys & allow_credentials).first
+        unless credential_id
+          raise "No matching credentials found for RP #{rp_id}"
```

Comment:
In an attempt to help debugging anyone using this option when testing, maybe we can be more verbose with the error and say `"Couldn't find credential with id=#{id} for RP=#{rp_id} among credentials=#{credentials[rp_id]}"`?

Reply:
Updated (not to the exact same message you suggested, but to a similar as a consequence of the thread below)!